lessons/Early-Math/distributive-property/langs/en-us/index.adoc

```diff
@@ -40,121 +40,264 @@ Students develop an understanding of the @vocab{Distributive Property} by examin
 
+
+@teacher{Without ample practice, there is a good chance that students will not find this process efficient. Like any solving strategy, it takes practice!}
+
 === Synthesize
 
 - How can you multiply two 2-digit numbers using mental math?
-- Can you think of a multiplication problem that would be _easier_ to solve using the Distributive Property?
-** _Note: We recommend inviting many students to share the problems they come up with. Make a list on the board for students to review together._
 - What sorts of problems are simpler to compute using the Distributive Property?
+- Can you think of a multiplication problem that would be _easier_ to solve using the Distributive Property?
+
+
+@teacher{We recommend inviting many students to share the problems they come up with. Make a list on the board for students to review together.}
+
+== Programming Exploration: Distribution @duration{20 minutes}
+
+=== Overview
+
+Extending concepts explored earlier in the lesson, students consider whether various functions that we use when coding are distributive.
+
+=== Launch
+
+We've learned a lot about how the distributive property works. We know that multiplication can distribute over addition or subtraction, allowing us to complete otherwise complex computations with efficiency and ease.
+
+@slidebreak
+
+But what about our programming functions? For instance, does `scale` distribute over `beside`?!
+
+@ifnotslide{
+[.embedded, cols="^.^4,^.^1,^.^4", stripes="none", grid="none", frame="none"]
+|===
+
+| @show{(coe '(scale 0.5 (beside aqua-star orange-dot)))} | &rarr; | @show{(coe  '(beside (scale 0.5 orange-dot) (scale 0.5 aqua-star)))}
+
+|===
+}
+
+@lesson-instruction{
+- After studying the two Circles of Evaluation (above), make a prediction: What image will the second Circle of Evaluation produce? Draw your prediction on @printable-exercise{distribution-and-code.adoc}.
+- Test your prediction. Type into @starter-file{editor} the code for the second Circle of Evaluation.
+}
+
+@slidebreak
+
+@QandA{
+@Q{Does `scale` distribute over `beside`? How do you know?}
+@A{Yes, it does. We know because both Circles of Evaluation produced identical images.}
+}
+
+@lesson-instruction{
+- Complete @printable-exercise{distribution-and-code.adoc}.
+}
+
+@QandA{
+@Q{What did you discover about the distributive property and code?}
+@Q{Why do you think `scale` distributed over `beside`, but `rotate` did not?}
+}
+
+
+=== Investigate
+
+Let's look at whether we can distribute two additional functions: `flip-vertical` and `flip-horizontal`.
+
+An interesting feature of these functions is that they require just *one* input: an Image. In a mathematics context, we have applied the Distributive Property to functions that require two inputs... but never one. Let's give it a try now!
+
+@slidebreak
+
+A function `triple` takes a single value and multiplies it by three. Study the pair of Circles of Evaluation below to see if we can distribute the function `triple` over addition.
```

Comment:
I like this a lot, but I think you can make it even stronger by introducing `triple` _before you get to programming_.

1. Tell them there's a made-up operation called triple, and you want to know if it can be distributed over addition. 
2. Then remind them of it here in the programming section

If we don't show that this is based _sound mathematics_, we'll have to deal with teachers who claim this is all just "programming stuff, not math"

Reply:
you're so right.